### ISSUE TITLE
bindings-all.py passing unknown params to bindgen

### ISSUE THREAD

**Issue by RReverser** [servo/rust-mozjs#363]
These params cause bindgen to throw errors:

```
  --no-class-constants \
  --no-type-renaming \
...
  --allow-unknown-types \
...
  --match ...
```

Could you clarify if these are simply outdated or you're using some private fork of bindgen with support for these? If the latter, any chance to send changes upstream so that we could all run bindings generator?

**Comment by jdm** [comment]
It's much more likely that the code in the tree is using an outdated version than a private fork.

**Comment by RReverser** [comment]
Yeah looks like that. For example, `allow-unknown-types`: https://github.com/servo/rust-bindgen/commit/8ed69e853f249e71c7b1f8f24b0297f2473d1e31

But then, there is no analogy for `--match` in up-to-date version, and the script seems to heavily rely on it to filter files instead of functions/types/...

**Comment by jdm** [comment]
There's ongoing work to integrate this work with upstream Spidermonkey automatically, which is really the future of this repository: https://bugzilla.mozilla.org/show_bug.cgi?id=1277338
